src/st/lib/e_dhuron.c

```diff
@@ -1,11 +1,210 @@
+                    !self->ext.dhuron.unk84) {
+                    SetSubStep(2);
+                }
+            }
+            if (!self->animFrameDuration && self->animFrameIdx == 6) {
+                PlaySfxPositional(SFX_BONE_SWORD_SWISH_B);
+            }
+            break;
+
+        case 2:
+            if (!AnimateEntity(D_us_801829F0, self)) {
+                SetStep(3);
+            }
+            break;
+        }
+        break;
+
+    case 5:
+        switch (self->step_s) {
+        case 0:
+            if (!AnimateEntity(D_us_801829A4, self)) {
+                self->ext.dhuron.unk84 = 0x20;
+                self->step_s++;
+            }
+            if (self->animFrameIdx == 10 && !self->animFrameDuration) {
+                PlaySfxPositional(SFX_THUNDER_B);
+                tempEntity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+                if (tempEntity != NULL) {
+                    CreateEntityFromEntity(E_ID_35, self, tempEntity);
+                    tempEntity->facingLeft = self->facingLeft;
+                    tempEntity->zPriority = self->zPriority - 1;
+                    tempEntity->ext.dhuron.unk9C = self;
+                }
+            }
+            break;
+
+        case 1:
+            if (!--self->ext.dhuron.unk84) {
+                self->ext.dhuron.unk89 = 1;
+                (self + 1)->ext.dhuron.unk89 = 1;
+                self->drawMode |= DRAW_TPAGE2 | DRAW_TPAGE;
+                SetStep(3);
+            }
+            break;
+        }
+        break;
+
+    case 8:
+        self->hitboxState = 0;
+        for (i = 0; i < 7; i++) {
+            tempEntity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+            if (tempEntity != NULL) {
+                CreateEntityFromEntity(E_ID_36, self, tempEntity);
+                tempEntity->params = i;
+                tempEntity->facingLeft = self->facingLeft;
+            } else {
+                break;
+            }
+        }
+        tempEntity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (tempEntity != NULL) {
+            CreateEntityFromEntity(E_EXPLOSION, self, tempEntity);
+            tempEntity->params = 3;
+        }
+        PlaySfxPositional(SFX_SKELETON_DEATH_A);
+        DestroyEntity(self);
+        return;
+    }
+    if (self->ext.dhuron.unk89) {
+        if (g_Timer & 2) {
+            self->palette = 0x201;
+        } else {
+            self->palette = 0x202;
+        }
+    }
+    hitboxPtr = D_us_801829FC[0];
+    tempVar = D_us_80182A14[self->animCurFrame];
+    hitboxPtr += tempVar * 4;
+    self->hitboxOffX = *hitboxPtr++;
+    self->hitboxOffY = *hitboxPtr++;
+    self->hitboxWidth = *hitboxPtr++;
+    self->hitboxHeight = *hitboxPtr++;
+}
 
 // Dhuron death parts
 INCLUDE_ASM("st/lib/nonmatchings/e_dhuron", func_us_801CC6B0);
 
 // Dhuron death
 INCLUDE_ASM("st/lib/nonmatchings/e_dhuron", func_us_801CC7BC);
+
+INCLUDE_ASM("st/lib/nonmatchings/e_dhuron", func_us_801CC984);
```

Comment:
Would be good to move this up for the PSP too if possible, otherwise I can do it a bit later on if you don't have it setup